autospec/specfiles.py

```diff
@@ -73,12 +75,40 @@ def __init__(self, url, version, name, release, config, requirements, content):
 
     def write_spec(self):
         """Write spec file."""
-        self.specfile = open_auto("{}/{}.spec".format(self.config.download_path, self.name), "w")
+        spec_path = f"{os.path.join(self.config.download_path, self.name)}.spec"
+        self.specfile = open_auto(spec_path, "w")
         self.specfile.write_strip = types.MethodType(_file_write, self.specfile)
 
+        # last change to sanitize url for template and build types
+        if self.config.urlban:
```

Comment:
Why does urlban become optional?

Reply:
I didn't think this changed anything logically, it just moved it from the nvr writing function. The reasoning for this to be optional is because the config for urlban isn't always existing (external builders probably don't use that config).

Reply:
OK, I guess we're reading it from common, not from the package options.conf.